kornia/losses/ms_ssim_l1.py

```diff
@@ -0,0 +1,150 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# Based on:
+# https://github.com/psyrocloud/MS-SSIM_L1_LOSS
+
+
+class MS_SSIM_L1Loss(nn.Module):
+    r"""Creates a criterion that computes MS-SSIM + L1 loss.
+
+    According to [1], we compute the MS-SSIM + L1 loss as follows:
+
+    .. math::
+        \text{loss}(x, y) = \alpha \cdot \mathcal{L_{MS-SSIM}}(x,y)+(1 - \alpha) \cdot G_\alpha \cdot \mathcal{L_1}(x,y)
+
+    Where:
+        - :math:`\alpha` is the weight parameter.
+        - :math:`x` and :math:`y` are the reconstructed and true reference images.
+        - :math:`\mathcal{L_{MS-SSIM}}` is the MS-SSIM loss.
+        - :math:`G_\alpha` is the sigma values for computing multi-scale SSIM.
+        - :math:`\mathcal{L_1}` is the L1 loss.
+
+    Reference:
+        [1]: https://research.nvidia.com/sites/default/files/pubs/2017-03_Loss-Functions-for/NN_ImgProc.pdf#page11
+
+    Args:
+        sigmas: gaussian sigma values.
+        data_range: the range of the images.
+        K: k values.
+        alpha : specifies the alpha value
+        compensation: specifies the scaling coefficient.
+
+    Returns:
+        The computed loss.
+
+    Shape:
+        - Input: :math:`(N, C, H, W)`.
+        - Output: :math:`(N,)` or scalar.
+
+    Examples:
+        >>> input1 = torch.rand(1, 3, 5, 5)
+        >>> input2 = torch.rand(1, 3, 5, 5)
+        >>> criterion = kornia.losses.MS_SSIM_L1Loss()
+        >>> loss = criterion(input1, input2)
+    """
+
+    def __init__(self,
+                 sigmas: list = [0.5, 1.0, 2.0, 4.0, 8.0],
+                 data_range: float = 1.0,
+                 K: list = [0.01, 0.03],
+                 alpha: float = 0.025,
+                 compensation: float = 200.0) -> None:
+        super().__init__()
+        self.DR = data_range
+        self.C1 = (K[0] * data_range) ** 2
+        self.C2 = (K[1] * data_range) ** 2
+        self.pad = int(2 * sigmas[-1])
+        self.alpha = alpha
+        self.compensation = compensation
+
+        # Set filter size
+        filter_size = int(4 * sigmas[-1] + 1)
+        g_masks = torch.zeros((3 * len(sigmas), 1, filter_size, filter_size))
+
+        # Compute mask at different scales
+        for idx, sigma in enumerate(sigmas):
+            g_masks[3 * idx + 0, 0, :, :] = self._fspecial_gauss_2d(filter_size, sigma)
+            g_masks[3 * idx + 1, 0, :, :] = self._fspecial_gauss_2d(filter_size, sigma)
+            g_masks[3 * idx + 2, 0, :, :] = self._fspecial_gauss_2d(filter_size, sigma)
+
+        self.g_masks = g_masks
+
+    def _fspecial_gauss_1d(self, size, sigma):
+        """Create 1-D gauss kernel
+        Args:
+            size (int): the size of gauss kernel
+            sigma (float): sigma of normal distribution
+        Returns:
+            torch.Tensor: 1D kernel (size)
+        """
+        coords = torch.arange(size).to(dtype=torch.float)
```

Comment:
```suggestion
        coords = torch.arange(size, device=device, dtype=dtype)
```
possibly when we move out this should be
`torch.arange(size, device=device, dtype=dtype)` where `device=torch.device('cpu')` and `dtype=torch.float32` in the function signature